sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVOptions.scala

```diff
@@ -150,6 +150,12 @@ class CSVOptions(
 
   val isCommentSet = this.comment != '\u0000'
 
+  /**
+   * The option enables checks of headers in csv files. In particular, column names
+   * are matched to field names of provided schema.
+   */
+  val checkHeader = getBool("checkHeader", true)
```

Comment:
Since the default is true, will it break the current app? This is a behavior change. We need to document it in the migration guide.

Reply:
Yes, it will break apps that declare explicitly schemas with field names different from header's column names like for example in a test of the PR that I modified. I will add info about the option to exception's message and update the migration guide.